inc/class-wpseo-image-utils.php

```diff
@@ -97,12 +106,14 @@ public static function has_usable_file_size( $image ) {
 	 */
 	public static function get_image( $attachment_id, $size ) {
 		if ( $size === 'full' ) {
-			$image         = wp_get_attachment_metadata( $attachment_id );
-			$image['url']  = wp_get_attachment_image_url( $attachment_id, 'full' );
-			$image['path'] = get_attached_file( $attachment_id );
+			$image = wp_get_attachment_metadata( $attachment_id );
```

Comment:
I want to propose extracting this into its own function `get_fullsize_image` which can either return the array or false if it's not an array.

This will also (possibly) help in regards to testing.